### ISSUE TITLE
Unable to Find ElementFromHandle in IE Mode - VBA

### ISSUE THREAD

**Issue by YN32** [longvh211/Edge-IE-Mode-Automation-with-IES-for-VBA#2]
I need to be able to find ElementFromHandle in order to select the appropriate tab.   Using Inspect.exe, I was able to verify that the handle found in the Findwindowex for the webpage was correct.  However, when I applied it to the function ElementFromHandle it gave an error "Automation error - An event was unable to invoke any of the subscribers".  Example

Dim UIAuto As New CUIAutomation 
Dim eleIEMode As IUIAutomationElemen 
Dim hndIEmode As LongPtr
Set UIAuto = New CUIAutomation
Set eleIEMode  = UIAuto.ElementFromHandle(ByVal hndIEmode)

Numerous examples posted on the website indicate that it is possible to get the ElementFromHandle for Chrome/Edge but it is all written in C++ .  I do not know C++.   Can it be done with VBA?

**Comment by longvh211** [comment]
Just to be clear: this IES framework does not involve the usage of UIA so this question is not related to the framework. Hence if you have any issue with applying UIA, it is likely that you are not using UIA correctly on the right object.

By the way, if you are looking to automate Edge IE Mode with UIA, why do you still need the IES framework?

**Comment by YN32** [comment]
Anh Long,
Appreciate it very much for taking the time to review my issue.
I understand the IES framework does not involve the usage of UIA.  However, with the original IE browser, I still need to use UIA to perform tasks such as downloading a file from webpage that has a prompt "Frame Notification Bar"
Thus, without UIA involvement, I am afraid IES framework is very limited to what I can do.
Anyway, you stated that I am likely are not using UIA correctly.  Please consider these codes that I found from John_W at ozgrid.com:
https://forum.ozgrid.com/forum/index.php?user-post-list/207280-john-w/&pageNo=1.   I tested these codes using IE and it works beautifully.  In my codes for the original IE browser, I do not have a need to use UIA to achieve what he did.  However, since IES framework is limited, I have to resort to UIA to overcome some of the issues.  Unfortunately, I am stuck at ElementFromHandle function.  Please look at line 86 in his code:

Set IEwindow = UIauto.ElementFromHandle(ByVal IEhwnd)

It works fine for the handle associated with IE but for edge or chrome, it did not recognize the handle.  Again, I used numerous API functions to test if the handle I got from edge/chrome is correct before I passed it to the function by using Inspect.exe.  I even applied the handle to the functions such as BringWindowToTop, ShowWindow, etc. and it works but not with the ElementFromHandle 
Again, any chance you can help to find out why I am unable to use ElementFromHandle function in VBA for edge/chrome handle?
r/dn 


    On Wednesday, August 31, 2022 at 08:36:09 AM PDT, Long Vh ***@***.***> wrote:  
 
 


Just to be clear: this IES framework does not involve the usage of UIA so this question is not related to the framework. Hence if you have any issue with applying UIA, it is likely that you are not using UIA correctly on the right object.

By the way, if you are looking to automate Edge IE Mode with UIA, why do you still need the IES framework?

**Comment by longvh211** [comment]
This is likely the case where the handle that you retrieve with your method may not be the correct handle for UIA due to complex window structure.

Do have a look at the new demo in the [release 1.1 ](https://github.com/longvh211/Edge-IE-Mode-Automation-with-IES-for-VBA/releases/tag/1.1) which I made the IE window handle exposed for UIA. Then give a try again with your case based on the example provided there.

**Comment by YN32** [comment]
Anh Long,
You DID it!  Not sure how you were able to do it but your demo work!  I have not applied it to my codes yet  I got excited and wanted to let you know the result.
will get back to you with another email.
THANK YOU!
r/dn


    On Thursday, September 1, 2022 at 03:34:08 AM PDT, Long Vh ***@***.***> wrote:  
 
 


This is likely the case where the handle that you retrieve may not be the correct handle for UIA. Do have a look at release 1.1 here  which I made the IE window handle exposed for UIA. Give a try to see if it works in your case.
